### DISCUSSION TITLE
External declaration must have an implementation in JS file

### DISCUSSION THREAD

**Discussion by rynowak** [microsoft/typespec#3631] [Q&A] [answered]
I'm trying to follow the tutorial [here](https://typespec.io/docs/extending-typespec/create-decorators) and getting stuck. 

<img width="855" alt="image" src="https://github.com/microsoft/typespec/assets/1430011/3dac57aa-6984-4bf6-8eab-6b62bc0af58c">

I've done my best to make sense of the [troubleshooting steps](https://typespec.io/docs/extending-typespec/create-decorators#troubleshooting), and I'm still stuck. I'd appreciate some help figuring out my mistake.

I don't understand this step in the troubleshooting guide so I've skipped it:

> You can use --trace bind.js.decorator to log debug information about decorator loading in the JS file, which should help identify the issue.

Some things that I've confirmd:

- I'm using `npm run watch` so my typescript changes are compiled.
- The function I declared on the typescript side is prefixed with `$` and exported.
- My `main.tsp` is importing the typescript output (`import "../dist/src/index.js";`)
- I'm using `setTypeSpecNamespace` to set the correct namespace on the typescript side.

Here is my repro: https://github.com/rynowak/udt-versioning/tree/decorator-repro

**Comment by bterlson** [comment, marked as answer]
Before I dig into the particulars (which I can likely do later today), can you confirm that you are getting this same error on the command line as well, or if the error still repros after restarting the language server? We have some issues with live updating ES Module bindings when you make changes to JS/TS source files.

**Reply by rynowak** [reply]
Looks like that did it! 

I can compile `lib/main.tsp` successfully at the command line: `tsp compile lib/main.tsp`. 

Also restarting vscode fixed the issue.

**Reply by bterlson** [reply]
You can also use the command palette "TypeSpec: restart language server" to avoid having to restart the entire program if you like!
